src/components/Home/presenter.js

```diff
@@ -115,6 +117,9 @@ class Home extends React.Component<Props, State> {
       return true;
     }
 
+    if (this.props.navigation != null) {
```

Comment:
The idea is reset to Home component instead of navigate to Home in SplashScreen. In most of the case, we don't need backhandler navigate to Splash.